inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/task/filecollect/FileScanner.java

```diff
@@ -0,0 +1,174 @@
+            recoverTime -= NewDateUtils.caclOffset(conf.getTimeOffset());
+        }
+
+        String startTime = NewDateUtils.millSecConvertToTimeStr(failTime, cycleUnit);
+        String endTime = NewDateUtils.millSecConvertToTimeStr(recoverTime, cycleUnit);
+        logger.info("task {} this scan time is between {} and {}.",
+                new Object[]{conf.getTaskId(), startTime, endTime});
+
+        return scanTaskBetweenTimes(conf, originPattern, startTime, endTime);
+    }
+
+    /* Scan log files and create tasks between two times. */
+    public static List<BasicFileInfo> scanTaskBetweenTimes(TaskProfile conf, String originPattern, String startTime,
+            String endTime) {
+        String cycleUnit = conf.getCycleUnit();
+        int maxFileNum = conf.getInt(TaskConstants.FILE_MAX_NUM);
+        List<Long> dateRegion = NewDateUtils.getDateRegion(startTime, endTime, cycleUnit);
+        List<BasicFileInfo> infos = new ArrayList<BasicFileInfo>();
+        for (Long time : dateRegion) {
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeInMillis(time);
+            String filename = NewDateUtils.replaceDateExpression(calendar, originPattern);
+            ArrayList<String> allPaths = FilePathUtil.cutDirectory(filename);
+            String firstDir = allPaths.get(0);
+            String secondDir = allPaths.get(0) + File.separator + allPaths.get(1);
+            ArrayList<String> fileList = getUpdatedOrNewFiles(firstDir, secondDir, filename, 3,
+                    maxFileNum);
+            for (String file : fileList) {
+                // TODO the time is not YYYYMMDDHH
+                String dataTime = NewDateUtils.millSecConvertToTimeStr(time, cycleUnit);
+                BasicFileInfo info = new BasicFileInfo(file, dataTime);
+                logger.info("scan new task fileName {} ,dataTime {}", file,
+                        NewDateUtils.millSecConvertToTimeStr(time, cycleUnit));
+                infos.add(info);
+            }
+        }
+        return infos;
+    }
+
+    public static ArrayList<String> scanFile(TaskProfile conf, String originPattern, long dataTime) {
+        int maxFileNum = conf.getInt(TaskConstants.FILE_MAX_NUM);
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTimeInMillis(dataTime);
+
+        String filename = NewDateUtils.replaceDateExpression(calendar, originPattern);
+        ArrayList<String> allPaths = FilePathUtil.cutDirectory(filename);
+        String firstDir = allPaths.get(0);
+        String secondDir = allPaths.get(0) + File.separator + allPaths.get(1);
+        return getUpdatedOrNewFiles(firstDir, secondDir, filename, 3, maxFileNum);
+    }
+
+    private static ArrayList<String> getUpdatedOrNewFiles(String firstDir, String secondDir,
+            String fileName, long depth, int maxFileNum) {
+
+        // logger.info("getUpdatedOrNewFiles: firstdir: {}, seconddir: {} filename: {}",
```

Comment:
It is recommended to delete the commented code

Reply:
get